plugins/docker/docker_

```diff
@@ -45,44 +46,179 @@ https://docs.docker.com/compose/reference/envvars/):
     env.DOCKER_HOST unix://var/run/docker.sock
 
 =back
+
+=head1 AUTHORS
+
+This section has been reverse-engineered from git logs
+
+* Codimp <contact@lithio.fr>: original rewrite
+* Rowan Wookey <admin@rwky.net>: performance improvement
+* Olivier Mehani <shtrom@ssji.net>: Network support, ClientWrapper, gerenal
+  cleanup
+
+=head1 MAGIC MARKERS
+
+#%# family=auto
+#%# capabilities=autoconf suggest
+
 """
 
 import os
 import sys
-import docker
+from functools import cached_property
 from multiprocessing import Process, Queue
 
+# Early autoconf-style detection
+try:
+    import docker
+except ImportError as e:
+    print(f'no ({e})')
+    sys.exit(1)
+
+
+def sorted_by_creation_date(func):
```

Comment:
Is this a good order of things?

After a reboot of the hypervisor machine, this would probably change the order of items (and their color) in the graph, or?

Maybe use the name instead?

(but feel free to follow your own judgement here - I cannot really tell)

Reply:
The name is generally random. The creation date doesn't change for the lifetime of the container.

The colours will always be a bit unstable, as they will change anytime you delete an old container, but at least, in this way, containers only get added at the end, rather than in between based on their names.